### ISSUE TITLE
The test suite xml file desnt"com.appium.executor.MyTestExecutor#constructXmlSuiteForParallel" method 

### ISSUE THREAD

**Issue by WenRui13** [saikrishna321/AppiumTestDistribution#308]

**Comment by WenRui13** [comment]
my pom file is bellow
 <dependency>
          <groupId>com.github.saikrishna321</groupId>
          <artifactId>AppiumTestDistribution</artifactId>
          <version>8.0.0</version>
      </dependency>

my config file is bellow
RUNNER=parallel
APPIUM_JS_PATH=/usr/local/lib/node_modules/appium/build/lib/main.js
BROWSER_TYPE=chrome
APP_TYPE=NA
BUNDLE_ID=com.xstore.sevenFresh
FRAMEWORK=testng
#CAPS=./caps/capabilities.json
DEVICE_PLATFORM=device

my Runner class  is bellow:
public class Runner {
    @Test
    public static void testApp() throws Exception {
        ExtentManager.setSystemInfoInReport("version","1.0.0");
        List<String> tests = new ArrayList<>();
        tests.add("ParallelDemoTest");

        ParallelThread parallelThread = new ParallelThread();
        boolean hasFailures = parallelThread.runner("com.xstore.qa",tests);
        Assert.assertFalse(hasFailures, "Testcases have failed in parallel execution");
    }
}

config as above , after running command:Platform="iOS" mvn clean -Dtest=Runner test
there is no test class contained in the target/parallel.xml  file,  see bellow:
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="2" verbose="2" name="TestNG Forum" parallel="tests" data-provider-thread-count="2">
  <listeners>
    <listener class-name="com.appium.manager.AppiumParallelTestListener"/>
    <listener class-name="com.appium.utils.RetryListener"/>
  </listeners>
  <test name="TestNG Test0" preserve-order="false">
    <parameter name="hostName" value="127.0.0.1"/>
    <parameter name="device" value="e4f98d12d6b570821e97e909c79af7f0590418e5"/>
  </test> 
  <test name="TestNG Test1" preserve-order="false">
    <parameter name="hostName" value="127.0.0.1"/>
    <parameter name="device" value="15cd5139e41348743248bcdc9e418bd7c78d40fb"/>
  </test> 
</suite> 


help me, thanks!

**Comment by SrinivasanTarget** [comment]
@WenRui13 Thanks for bringing this to our notice and will look into this soon.

**Comment by WenRui13** [comment]
the implementation in the dependency source class is bellow:
```java
    private List<XmlClass> writeXmlClass(List<String> testcases, Map<String, List<Method>> methods, List<XmlClass> xmlClasses) {
        Iterator var4 = methods.keySet().iterator();

        while(true) {
            while(true) {
                String className;
                do {
                    if (!var4.hasNext()) {
                        return xmlClasses;
                    }

                    className = (String)var4.next();
                } while(!className.contains("Test"));

                if (testcases.size() == 0) {
                    xmlClasses.add(this.createClass(className));
                } else {
                    Iterator var6 = testcases.iterator();

                    while(var6.hasNext()) {
                        String s = (String)var6.next();

                        for(int j = 0; j < this.items.size(); ++j) {
                            String testName = ((String)this.items.get(j)).concat("." + s).toString();
                            if (testName.equals(className)) {
                                xmlClasses.add(this.createClass(className));
                            }
                        }
                    }
                }
            }
        }
    }
```
but in the cloned project I see is：
```java
    private List<XmlClass> writeXmlClass(List<String> testcases, Map<String,
            List<Method>> methods, List<XmlClass> xmlClasses) {
        for (String className : methods.keySet()) {
            if (className.contains("Test")) {
                if (testcases.size() == 0) {
                    xmlClasses.add(createClass(className));
                } else {
                    for (String s : testcases) {
                        for (int j = 0; j < items.size(); j++) {
                            String testName = items.get(j).concat("." + s).toString();
                            if (testName.equals(className)) {
                                xmlClasses.add(createClass(className));
                            }
                        }
                    }
                }

            }
        }
        return xmlClasses;
    }
```

**Comment by saikrishna321** [comment]
make sure classes files inside the package mentioned in the runner ends with Test as per maven/gradle standards

**Comment by WenRui13** [comment]
@saikrishna321 yes, I did it. but still can't get the right test suite file.

**Comment by saikrishna321** [comment]
@WenRui13 do u also have ParallelDemoTest class in the same package. Can you please share screenshot of your project in intellji

**Comment by saikrishna321** [comment]
@WenRui13  works fine on my side.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="1" verbose="2" name="TestNG Forum" parallel="tests" data-provider-thread-count="1">
  <listeners>
    <listener class-name="com.appium.manager.AppiumParallelTestListener"/>
    <listener class-name="com.appium.utils.RetryListener"/>
  </listeners>
  <test name="TestNG Test0" preserve-order="false">
    <parameter name="hostName" value="127.0.0.1"/>
    <parameter name="device" value="emulator-5554"/>
    <classes>
      <class name="com.test.site.DragnDropTest"/>
      <class name="com.test.site.SliderTest"/>
    </classes>
  </test> <!-- TestNG Test0 -->
</suite> <!-- TestNG Forum -->

```

**Comment by WenRui13** [comment]
of course. my project structure is 
![image](https://user-images.githubusercontent.com/6370420/37809515-bdfacc50-2e8a-11e8-8e94-0d395c9d6263.png)

**Comment by saikrishna321** [comment]
can u pls give same project to reproduce this

**Comment by WenRui13** [comment]
@saikrishna321 yes ,this is my project [https://github.com/WenRui13/sevenfresh-ios-test](https://github.com/WenRui13/sevenfresh-ios-test) to reproduce this  issue. pls correct it if there is some  wrong configurations. tks a lot!

**Comment by saikrishna321** [comment]
@WenRui13 thanks for the sample repo. @SrinivasanTarget can u pls try to reproduce this, as i need a real device. It works fine with simulator and android devices attached. 

Can you pls try having only bundleID

**Comment by WenRui13** [comment]
As above my mentioned , the  writeXmlClass  method in dependency  is different from your  source code whith verison 8.0.0,  Is it possible that this caused this issue?
 try having only bundleID, still the  same. Looking forward your great job，celebration work cheerful !

**Comment by SrinivasanTarget** [comment]
@WenRui13 Can you invalidate cache and restart. Also try reimporting all maven plugins with latest dependency.

**Comment by WenRui13** [comment]
@SrinivasanTarget yes , I do that ,still the same.

**Comment by SudhiHathwar** [comment]
Hi @WenRui13,

I can see there were two packages with same name (com.xstore.qa) one was under src/main/java and the other package was src/main/test.

I tried after deleting / updating the package name under main/java and it worked fine for me.

Can you please try the same?

@saikrishna321 @SrinivasanTarget
